### ISSUE TITLE
Trigger the mask after clone

### ISSUE THREAD

**Issue by Ihabafia** [igorescobar/jQuery-Mask-Plugin#434]
Hi,

I am cloning an HTML fields that they are masked,  the fields are contact information, so if the user click a button to add an extra contact he gets a set of fields one of them is the mobile number.

The cloned fields are not masked for some reason, so I am looking for a way to trigger the mask plug-in again to make that cloned field masked again.

I hope that I was able to explain it.

Here is an example:
```

<div class="row contact-row hidden">
	<div class="col col-sm-12">
		<div class="header line lighter bigger"></div>
		<div class="col-sm-12 toolRemove text-right hidden">
			<a href="#" data-action="collapse">
				<i class="ace-icon fa fa-minus" data-icon-show="fa-minus" data-icon-hide="fa-minus"></i>
			</a>
		</div>
		<div class="contacts">
			{# $required = in_array('contact_title_id', $requiredFields) ? "<span class='input-group-addon input-sm required'>*</span>" : "" #}
			{# $x = $required ? 'group' : 'field' #}
			<div class="col col-sm-3 form-group">
				{{ form_label(t('contactTitle'), 'contact_title_id', array('class' => 'control-label')) }}
				<div class="input-{{$x}}">
					@if($readOnly)
						<p id="contact_title_id" class="form-control-static bold"> {{(!empty($contacts['contact_title_id'])) ? $titlesOptions[$contacts['contact_title_id']]: '' }}</p>
					@else
						{{ form_dropdown('contacts[contact_title_id][]', $titlesOptions, set_value('contacts[contact_title_id][]', $contacts['contact_title_id']), 'class="form-control noSelect2" data-lang="'.lang().'"') }}
					@endif
					{{ $required }}
				</div>
			</div>
			<div class="col col-sm-9">
				{# $required = in_array('contact_name_en', $requiredFields) ? "<span class='input-group-addon input-sm required'>*</span>" : "" #}
				{# $x = $required ? 'group' : 'field' #}
				<div class="col col-sm-6 form-group">
					{{ form_label(t('contact_name_en'), 'contact_name_en', array('class' => 'control-label')) }}
					<div class="input-{{$x}}">
						@if($readOnly)
							<p id="contact_name_en" class="form-control-static bold"> {{(!empty($contacts['contact_name_en'])) ? $contacts['contact_name_en']: '' }}</p>
						@else
							{{ form_input('contacts[contact_name_en][]', set_value('contacts[contact_name_en][]', $contacts['contact_name_en']), ' class="form-control input-sm" placeholder="' . t("contact_name_enPH") . '"  ') }}
						@endif
						{{ $required }}
					</div>
				</div>
				{# $required = in_array('contact_name_ar', $requiredFields) ? "<span class='input-group-addon input-sm required'>*</span>" : "" #}
				{# $x = $required ? 'group' : 'field' #}
				<div class="col col-sm-6 form-group">
					{{ form_label(t('contact_name_ar'), 'contact_name_ar', array('class' => 'control-label')) }}
					<div class="input-{{$x}}">
						@if($readOnly)
							<p id="contact_name_ar" class="form-control-static bold"> {{(!empty($contacts['contact_name_ar'])) ? $contacts['contact_name_ar']: '' }}</p>
						@else
							{{ form_input('contacts[contact_name_ar][]', set_value('contacts[contact_name_ar][]', $contacts['contact_name_ar']), 'class="form-control input-sm" placeholder="' . t("contact_name_arPH") . '"  ') }}
						@endif
						{{ $required }}
					</div>
				</div>
			</div>
			{# $required = in_array('contact_mobile', $requiredFields) ? "<span class='input-group-addon input-sm required'>*</span>" : "" #}
			{# $x = $required ? 'group' : 'field' #}
			<div class="col-sm-6 form-group">
				{{ form_label(t('contact_mobile'), 'contact_mobile', array('class' => 'control-label')) }}
				<div class="input-{{$x}}">
					@if($readOnly)
						<p id="contact_mobile" class="form-control-static bold"> {{(!empty($contacts['contact_mobile'])) ? $contacts['contact_mobile']: '' }}</p>
					@else
						{{ form_input('contacts[contact_mobile][]', set_value('contacts[contact_mobile][]', ''), 'class="contact_mobile form-control input-sm mask unmask" placeholder="' . t("contact_mobilePH") . '"  ') }}
					@endif
					{{ $required }}
				</div>
			</div>
			{# $required = in_array('contact_email', $requiredFields) ? "<span class='input-group-addon input-sm required'>*</span>" : "" #}
			{# $x = $required ? 'group' : 'field' #}
			<div class="col-sm-6 form-group">
				{{ form_label(t('contact_email'), 'contact_email', array('class' => 'control-label')) }}
				<div class="input-{{$x}}">
					@if($readOnly)
						<p id="contact_email" class="form-control-static bold"> {{(!empty($contacts['contact_email'])) ? $contacts['contact_email']: '' }}</p>
					@else
						{{ form_input('contacts[contact_email][]', set_value('contacts[contact_email][]', $contacts['contact_email']), 'id="contact_emailI" class="form-control input-sm" placeholder="' . t("contact_emailPH") . '"  ') }}
					@endif
					{{ $required }}
				</div>
			</div>
		</div>
	</div>
</div>
```

Here is the jquery:

```
$('.contact_mobile').mask("+(000) 00 000-0000");

if(typeof CONTACT_ROW === 'undefined'){
	var CONTACT_ROW = $('div.contact-row.hidden:first');
	lang = $('select.noSelect2').data('lang');
	CONTACT_ROW.find('.toolRemove').removeClass('hidden');
	$('div.contact-row.hidden:first').detach();
};

$('#add').on("click", function() {
	if(counter > 3){
		$('a.add').on('click', function(event) {
			event.preventDefault();
			return false;
		});
	} else {
		var newRow = CONTACT_ROW.clone(true);
		newRow.removeClass('hidden');

        newRow.find('.noSelect2').addClass('dd-select_'+lang);
        newRow.find('.dd-select_'+lang).removeClass('noSelect2');
        newRow.find('input.contact_mobile').mask("+(000) 00 000-0000").removeClass('mask');

		$('div.contact-row:last').after(newRow);
		activate_select2();
		counter++;

		return false;
	}
})
```

**Comment by igorescobar** [comment]
@Ihabafia Don't need to clone it. Just add new fields and re-use the same classes. 

Take a look into our global options:
```javascript
// nonInput: elements we consider nonInput
// dataMask: we mask data-mask elements by default
// watchInputs: watch for dynamically added inputs by default
// watchDataMask: by default we disabled the watcher for dynamically added data-mask elements by default (performance reasons)
 $.jMaskGlobals = {
    maskElements: 'input,td,span,div',
    dataMaskAttr: '*[data-mask]',
    dataMask: true,
    watchInterval: 300,
    watchInputs: true,
    watchDataMask: false,
    byPassKeys: [9, 16, 17, 18, 36, 37, 38, 39, 40, 91],
    translation: {
        '0': {pattern: /\d/},
        '9': {pattern: /\d/, optional: true},
        '#': {pattern: /\d/, recursive: true},
        'A': {pattern: /[a-zA-Z0-9]/},
        'S': {pattern: /[a-zA-Z]/}
    }
  };
```
